components/channel_header/channel_header.tsx

```diff
@@ -487,7 +493,7 @@ class ChannelHeader extends React.PureComponent {
         }
 
         let headerTextContainer;
-        const headerText = (isDirect && dmUser.is_bot) ? dmUser.bot_description : channel.header;
+        const headerText = (isDirect && dmUser?.is_bot) ? dmUser.bot_description : channel.header;
```

Comment:
Similarly, should this be nullable, or is this just a case where we've null checked this above and TS can't identify that? It seems like we would've encountered errors before if this could actually be null

Reply:
If you are referring to `dmUser`, it should be nullable. Because
- Within `channel_header.tsx`, We are not null checking before assigning `headerText` 
- Within `index.ts`, `dmUser` is set under certain condition as shown below
https://github.com/mattermost/mattermost-webapp/blob/69d1015f95a3561c780991f6383a8a9923cd8678/components/channel_header/index.js#L65-L74

Reply:
I guess that's a case where we can say it should be true as long as `isDirect` is true, but there's no way to check that statically since they're not directly related